src/Symfony/Component/Mime/Tests/Part/SMimePartTest.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mime\Tests\Part;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Mime\Header\Headers;
+use Symfony\Component\Mime\Header\ParameterizedHeader;
+use Symfony\Component\Mime\Part\SMimePart;
+
+class SMimePartTest extends TestCase
+{
+    public function testConstructor()
+    {
+        $p = new SMimePart('content', 'multipart', 'signed', []);
+        $this->assertEquals('content', $p->bodyToString());
+        $this->assertEquals('content', implode('', iterator_to_array($p->bodyToIterable())));
+        // bodyToIterable() can be called several times
+        $this->assertEquals('content', implode('', iterator_to_array($p->bodyToIterable())));
+        $this->assertEquals('multipart', $p->getMediaType());
+        $this->assertEquals('signed', $p->getMediaSubType());
+    }
+
+    public function testConstructorWithIterable()
+    {
+        $iterable = $this->getIterable();
+
+        $p = new SMimePart($iterable, 'multipart', 'signed', []);
+        $this->assertEquals('content', $p->bodyToString());
+        $this->assertEquals('content', implode('', iterator_to_array($p->bodyToIterable())));
+        // bodyToIterable() can be called several times
+        $this->assertEquals('content', implode('', iterator_to_array($p->bodyToIterable())));
+        $this->assertEquals('multipart', $p->getMediaType());
+        $this->assertEquals('signed', $p->getMediaSubType());
+    }
+
+    public function testConstructorWithInvalidIterable()
+    {
+        $iterable = $this->getIterable();
+        // We using this method for close a generator which should throws: Exception : Cannot traverse an already closed generator
+        iterator_to_array($iterable);
+
+        $p = new SMimePart($iterable, 'multipart', 'signed', []);
+        $this->assertEquals('', $p->bodyToString());
+        $this->assertEquals('', implode('', iterator_to_array($p->bodyToIterable())));
```

Comment:
Should this throw an exception instead of silently return an empty string?
What if we have the following test

```
 $iterable = $this->getIterable();

$p = new SMimePart($iterable, 'multipart', 'signed', []);
$this->assertEquals('content', implode('', iterator_to_array($p->bodyToIterable())));
$this->assertEquals('', implode('', iterator_to_array($p->bodyToIterable())));
```

This looks weird to me.

Reply:
The result always will be `content` because `bodyToIterable` always generate newest object of Generator which is not closed and still valid.